src/BuildLoop.cpp

```diff
@@ -82,11 +73,15 @@ static void LogFirstTimeEnqueue(MemAllocLinear* scratch, RuntimeNode* enqueuedNo
     LogStructured(&msg);
 }
 
-static int EnqueueNodeListWithoutWakingAwaiters(BuildQueue* queue, MemAllocLinear* scratch, const FrozenArray<int32_t>& nodesToEnqueue, RuntimeNode* enqueingNode)
+static int EnqueueNodeListReversedWithoutWakingAwaiters(BuildQueue* queue, MemAllocLinear* scratch, const FrozenArray<int32_t>& nodesToEnqueue, RuntimeNode* enqueingNode)
 {
     int enqueue_count = 0;
-    for (int32_t depDagIndex : nodesToEnqueue)
+
+    //we enqueue dependency lists in reverse. because our semantics say that the most urgent dependencies are in the list first, it's important that they
+    //end up on the top of the workstack, so they'll be processed first.
+    for(int i=nodesToEnqueue.GetCount(); i>=0; i++)
```

Comment:
Wait, how does this loop work? Won't it run forever, since we start with a value >0, and then run as long as it's >0, but only increment. Should we do `i--`?